File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        triangle = [[1]]
        arr = [1]
        for i in range (2, rowIndex + 2):
            arr = [0] * i
            arr[0] = 1
            arr[i - 1] = 1
            for j in range(1, i - 1):
                arr[j] = triangle[i - 2][j - 1] +  triangle[i - 2][j]
            triangle.append(arr)
        return arr

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> prevRow;
        List<Integer> currRow = new ArrayList<>();
        int row = 0;
        currRow.add(1);
        while (row < rowIndex) {
            row++;
            prevRow = currRow;
            currRow = new ArrayList<>();
            currRow.add(1);
            for (int i = 1; i < row; i++) {
                currRow.add(prevRow.get(i - 1) + prevRow.get(i));
            }
            currRow.add(1);
        }
        
        return currRow;
    }
}